lambdas/tests/unit/services/test_document_service.py

```diff
@@ -30,59 +31,74 @@ def nhs_number():
+    result = service.fetch_available_document_references_by_type(TEST_NHS_NUMBER, "LG")
 
-        assert len(result) == 3
-        assert type(result[0]) == DocumentReference
+    mock_dynamo_table.assert_called_with(MOCK_LG_TABLE_NAME)
 
+    assert len(result) == 3
+    assert type(result[0]) == DocumentReference
 
-def test_only_retrieves_documents_from_lloyd_george_table(nhs_number):
-    expected_table = "expected_table_name"
-    os.environ["LLOYD_GEORGE_DYNAMODB_NAME"] = expected_table
-    os.environ["DOCUMENT_STORE_DYNAMODB_NAME"] = "no-table"
 
-    with patch.object(boto3, "resource", return_value=MagicMock()) as mock_dynamo:
-        mock_table = MagicMock()
-        mock_dynamo.return_value.Table.return_value = mock_table
-        mock_table.query.return_value = MOCK_EMPTY_RESPONSE
-        result = DocumentService().fetch_available_document_references_by_type(
-            nhs_number, "LG"
-        )
+def test_returns_list_of_arf_document_references_when_results_are_returned(
+    set_env, mocker
+):
+    mocker.patch("boto3.resource")
+    service = DocumentService()
 
-        mock_dynamo.return_value.Table.assert_called_with(expected_table)
+    mock_dynamo_table = mocker.patch.object(service.dynamodb, "Table")
+    mock_dynamo_table.return_value.query.return_value = MOCK_SEARCH_RESPONSE
 
-        assert len(result) == 0
+    result = service.fetch_available_document_references_by_type(TEST_NHS_NUMBER, "ARF")
 
+    mock_dynamo_table.assert_called_with(MOCK_ARF_TABLE_NAME)
 
-def test_only_retrieves_documents_from_electronic_health_record_table(nhs_number):
-    expected_table = "expected_table_name"
-    os.environ["LLOYD_GEORGE_DYNAMODB_NAME"] = "no-table"
-    os.environ["DOCUMENT_STORE_DYNAMODB_NAME"] = expected_table
+    assert len(result) == 3
+    assert type(result[0]) == DocumentReference
 
-    with patch.object(boto3, "resource", return_value=MagicMock()) as mock_dynamo:
-        mock_table = MagicMock()
-        mock_dynamo.return_value.Table.return_value = mock_table
-        mock_table.query.return_value = MOCK_EMPTY_RESPONSE
-        result = DocumentService().fetch_available_document_references_by_type(
-            nhs_number, "ARF"
-        )
 
-        mock_dynamo.return_value.Table.assert_called_with(expected_table)
+def test_returns_list_of_both_type_document_references_when_results_are_returned(
+    set_env, mocker
+):
+    mocker.patch("boto3.resource")
+    service = DocumentService()
 
-        assert len(result) == 0
+    mock_dynamo_table = mocker.patch.object(service.dynamodb, "Table")
+    mock_dynamo_table.return_value.query.side_effect = [
+        MOCK_SEARCH_RESPONSE,
+        MOCK_SEARCH_RESPONSE,
+    ]
+
+    result = service.fetch_available_document_references_by_type(TEST_NHS_NUMBER, "ALL")
+
+    assert mock_dynamo_table.call_count == 2
+
+    assert len(result) == 6
+    assert type(result[0]) == DocumentReference
+
+
```

Comment:
suggest that we can add a unit test when we fetching ALL but one type have record and the other type is empty response:
 
```suggestion

def test_returns_list_of_both_type_document_references_when_only_one_type_have_result(
    set_env, mocker
):
    mocker.patch("boto3.resource")
    service = DocumentService()

    mock_dynamo_table = mocker.patch.object(service.dynamodb, "Table")
    mock_dynamo_table.return_value.query.side_effect = [
        MOCK_SEARCH_RESPONSE,
        MOCK_EMPTY_RESPONSE,
    ]

    result = service.fetch_available_document_references_by_type(TEST_NHS_NUMBER, "ALL")

    assert mock_dynamo_table.call_count == 2

    assert len(result) == 3
    assert type(result[0]) is DocumentReference

```